File: apps/backend/competition-scraper/scraper/infrastructure/blockout/pool.py
from dataclasses import dataclass
from datetime import datetime


@dataclass(frozen=True)
class CreatePoolInternalRequest:
    """Exact request accepted by pools-service's create route."""

    poolCode: str
    leagueCode: str
    season: str
    leagueName: str
    rawName: str
    name: str
    shortName: str
    divisionId: int
    format: str
    gender: str
    followersCount: int | None
    active: bool


@dataclass(frozen=True)
class UpdatePoolInternalRequest:
    """Exact request accepted by pools-service's update route."""

    poolCode: str
    leagueCode: str
    season: str
    leagueName: str
    rawName: str
    name: str
    shortName: str
    divisionId: int
    format: str
    gender: str
    active: bool


@dataclass
class PoolInternalResponse:
    """Complete pools-service representation used by scraper decisions."""

    poolCode: str
    leagueCode: str
    season: str
    divisionId: int
    leagueName: str
    rawName: str
    name: str
    shortName: str
    format: str
    gender: str
    followersCount: int | None = 0
    active: bool = True
    id: int | None = None
    createdAt: datetime | None = None
    lastUpdate: datetime | None = None

File: apps/frontend/mobile/src/modules/pool/model/Pool.ts
import type { DivisionResponse } from "@/src/modules/division/model/Division";
import type { EnumFormat } from "@/src/shared/model/enums/Format";
import type { EnumGender } from "@/src/shared/model/enums/Gender";
import type { TeamWithStatsResponse } from "@/src/modules/team/model/Team";

export type PoolInternalResponse = {
  id: number;
  poolCode: string;
  leagueCode: string;
  season: string;
  divisionId: number;
  gender: EnumGender;
  format: EnumFormat;
  leagueName: string;
  rawName: string;
  name: string;
  shortName: string;
  followersCount: number;
  active: boolean;
  createdAt: string;
  lastUpdate: string;
};

export type UpdatePoolRequest = {
  poolCode?: string;
  leagueCode?: string;
  season?: string;
  leagueName?: string;
  rawName?: string;
  name?: string;
  shortName?: string;
  divisionId?: number;
  format?: EnumFormat;
  gender?: EnumGender;
  active?: boolean;
};

export type PoolResponse = {
  id: number;
  season: string;
  leagueCode: string;
  leagueName: string;
  poolCode: string;
  name: string;
  shortName: string;
  rawName: string;
  format: EnumFormat;
  gender: EnumGender;
  followersCount: number;
  ranking: TeamWithStatsResponse[];
  division: DivisionResponse;
};

export type PoolSummaryResponse = {
  id: number;
  name: string;
  shortName: string;
  season: string;
  gender: EnumGender;
  format: EnumFormat;
  division: DivisionResponse;
  leagueCode: string;
  leagueName: string;
};
